enhancements/fbc-run-bundle.md

```diff
@@ -0,0 +1,268 @@
+---
+title: fbc-run-bundle
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@joelanford"
+  - "@jmrodri"
+  - "@VenkatRamaraju"
+approvers:
+  - "@joelanford"
+  - "@jmrodri"
+  - "@VenkatRamaraju"
+creation-date: 2022-04-20
+last-updated: 2022-04-20
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Add FBC support to run bundle
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+- Do we need to build a File-Based Catalog to run this command?
+
+  - We can use `quay.io/operatorhubio/catalog:latest` for testing and generate a File-Based Catalog at runtime when the run bundle command is invoked.
+
+- Do we need to make our own image to use as a default one for run bundle and run bundle-upgrade?
+
+  - Currently, we are using `quay.io/operator-framework/opm:latest` as the default index image. We might have to build an image similar to `quay.io/operatorframework/catalog:latest` which uses opm as the base image and also has a built catalog in it.
+
+- While initializing the FBC, should the channel received from the bundle annotations be used to initialize the default channel name in the `opm init` command?
+
+  - We should be able to handle bundles that don't have channel metadata on the annotations.
+
+- If we don’t get channel information from the bundle, then can we add it to a default channel or to the channel HEAD?
```

Comment:
Do we have an answer to this question?

Reply:
What means channel HEAD?
Have we not only a channel that is the default package?

Reply:
According to the doc, the channel seems to be optional, but if we try to use bundle validate with the info empty "it fails. See: https://github.com/operator-framework/operator-sdk/issues/5569

Did we figure out if informing the channel should or not be mandatory?
Becuase I think that should answer this question.

Reply:
Also, how are we dealing with this scenario currently?
Why do we need to change the current behaviour? I mean, should we not deal with in the same way that we do now for SQLIndex? Why it is an open question under the scope to support FBC catalog index?

Reply:
@jmrodri @camilamacedo86, We use the channel that is specified in the bundle labels [here](https://github.com/operator-framework/operator-sdk/pull/5671/files#diff-2a945a0659f7f5a30cfc34d537d198566fd96e884bb783c8fd0190840bc87cc2R128).

This came up in the reviews, see the comment - https://github.com/operator-framework/operator-sdk/pull/5671#discussion_r853292644. The reason we are changing this up is because the bundle will no longer have channel metadata, see comment https://github.com/operator-framework/operator-sdk/pull/5671#discussion_r853292644.

Reply:
For FBC, channel membership and package-level metadata embedded in bundles is ignored. So we should design FBC tooling to ignore that as well.

I think I commented in a code PR that we should either hardcode (e.g. channel name = `develop`) or derive a channel name (e.g. channel name = `<package-name>-develop`). Seems like putting package name in the channel name is redundant though because the channel is already a member of the package.

Reply:
@joelanford Thanks for the insight. I've clarified this question on the EP in the last commit.